File: server/piwars-common/src/main/java/org/ah/piwars/virtualrover/game/rovers/GCCRoverM16.java
package org.ah.piwars.virtualrover.game.rovers;

import com.badlogic.gdx.math.Quaternion;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.math.Vector3;

import org.ah.themvsus.engine.common.game.GameObjectFactory;

import static org.ah.piwars.virtualrover.engine.utils.CollisionUtils.asShape2DList;
import static org.ah.piwars.virtualrover.engine.utils.CollisionUtils.polygonFromBox;

public class GCCRoverM16 extends Rover {

    public GCCRoverM16(GameObjectFactory factory, int id) {
        super(factory, id, RoverType.GCCM16);

        this.polygons = asShape2DList(polygonFromBox(-80f,  -55f, 80f,  55f));
        this.attachmentPosition = new Vector2(80f, 0);
        this.cameraPosition = new Vector3(-40f, 0f, 110f);
        this.cameraOrientation = new Quaternion();
        this.cameraOrientation.setEulerAngles(22.5f, 0f, 0f); // positive yaw is down
        this.cameraAngle = 45f;
    }
}

File: simulator/src/piwarssim/engine/simulation/rovers/GCCRoverM16.py

from enum import Enum

from piwarssim.engine.simulation.rovers.AbstractRoverSimObject import AbstractRoverSimObject
from piwarssim.engine.simulation.rovers.RoverType import RoverType


class GCCRoverM16(AbstractRoverSimObject):

    def __init__(self, factory, sim_object_id, sim_object_type):
        super(GCCRoverM16, self).__init__(factory, sim_object_id, sim_object_type, RoverType.GCCRoverM16)
        self.attachment_position = (80.0, 0.0)
        self.camera_position = (75.0, 0.0, 20.0)
        self.camera_orientation = AbstractRoverSimObject.from_yaw_pitch_roll(10.0, 0.0, 0.0) # positive yaw is down
        self.camera_angle = 45.0

    def __repr__(self):
        return "GCCRover[" + super(GCCRoverM16, self).__repr__() + "]"
